File: SimulaBarramentoAmil/controllers/ite003.js
module.exports = function(app) {

        app.get('/venda-online/api/corretor/:cpf/validacep/:cep', function(req, res) {

            var response = {
                ite003Response:{
                    codResponse: "ITE003-0000"
                }
            };

            console.log(req.headers);
            console.log('Request ITE - 003 : ' + JSON.stringify(req.params));

            if(req.header('authorization') === "Bearer 104bf8d3-d4c1-461c-9139-c565635010a3") {

                if (req.params.cep == undefined) {
                    response.ite003Response.codResponse = "ITE003-0001";
                    response.ite003Response.descCodResponse = "Mensagem Invalida";
                    console.log("cep invalido *******");
                } else if (req.params.cpf == undefined) {
                    response.ite003Response.codResponse = "ITE003-0001";
                    response.ite003Response.descCodResponse = "Mensagem Invalida";
                    console.log("cpf invalido *******");
                } else {
                    response.ite003Response.nomeCorretor = "Teste Amil"
                    response.ite003Response.codResponse = "ITE003-0000";
                    response.ite003Response.descCodResponse = "Mensagem Valida";
                }
                res.send(response);
            }else{
                res.send(response);
            }

        });
};

File: SendToBarramentoAmil/src/main/java/SendToBarramentoAmil/ITE003.java
package SendToBarramentoAmil;


import com.sun.jersey.api.client.Client;
import com.sun.jersey.api.client.ClientResponse;
import com.sun.jersey.api.client.WebResource;
import org.json.JSONObject;

/**
 * Created by lrafael on 24/07/17.
 */
public class ITE003 {

    public static String sendIte003(String body) throws Exception{

        String output = "";
        String auth = "";

        try {

            JSONObject jsonObject = new JSONObject(body).getJSONObject("ite003Request");
            auth = "Bearer " + jsonObject.getString("token");

            Client client = Client.create();

            WebResource webResource = client
                    .resource("https://api-hom.amil.com.br/venda-online/api/corretor/" + jsonObject.getString("cep") + "/validacep/" + jsonObject.getString("cpf"));

            ClientResponse response = webResource.type("application/x-www-form-urlencoded")
                    .header("Content-Type", "application/x-www-form-urlencoded")
                    .header("Authorization", auth)
                    .get(ClientResponse.class);

            int res = response.getStatus();
            output = response.getEntity(String.class);
        }catch (Exception e){
            throw e;
        }
      return output;
    };

}
